### PR TITLE
chore(core): bump next-intl

### PR DESCRIPTION
**Pull request by jorgemoya**
## What/Why?
+ Bump `next-intl`.
+ Use new `createNavigation` api.
+ Pass `locale` to `redirects`.
+ `setRequestLocale` is no longer unstable.
+ Fixes https://github.com/amannn/next-intl/issues/1335

## Testing
Locally, redirecting on default locale will not append the default locale, just like Links.
Updated tests to match this behavior.

### CODE REVIEW

File: core/app/admin/route.ts [hunk 1]
```diff
@@ -1,20 +1,25 @@
+import { useLocale } from 'next-intl';
+
 import { redirect } from '~/i18n/routing';
 
 const canonicalDomain: string = process.env.BIGCOMMERCE_GRAPHQL_API_DOMAIN ?? 'mybigcommerce.com';
 const BIGCOMMERCE_STORE_HASH = process.env.BIGCOMMERCE_STORE_HASH;
 const ENABLE_ADMIN_ROUTE = process.env.ENABLE_ADMIN_ROUTE;
 
 export const GET = () => {
+  const locale = useLocale();
```
Comment (deini) [inline core/app/admin/route.ts:10]: Can we use `getLocale` here?

File: core/app/xmlsitemap.php/route.ts [hunk 2]
```diff
@@ -8,6 +10,10 @@ import { permanentRedirect } from '~/i18n/routing';
  * on /xmlsitemap.php
  */
 
-export const GET = () => permanentRedirect('/sitemap.xml');
+export const GET = () => {
+  const locale = useLocale();
```
Comment (chanceaclark) [inline core/app/xmlsitemap.php/route.ts:14]: Same here

**Comment by jorgemoya** [comment]
Reported this issue from working on this: https://github.com/amannn/next-intl/issues/1438

**Comment by jorgemoya** [comment]
I had to pass default locale for these redirect in routes since next-intl does not work here.

**Review by chanceaclark — APPROVED** [review]

**Review by deini — APPROVED** [review]

**Pull request merged**